moped-editor/src/views/projects/projectView/ProjectFunding/ProjectFundingTable.js

```diff
@@ -130,9 +106,164 @@ const useFdusArray = (projectFunding) =>
+        renderEditCell: (props) => (
+          <LookupSelectComponent
+            {...props}
+            name={"funding_status"}
+            defaultValue={1}
+            data={data.moped_fund_status}
+          />
+        ),
+      },
+      {
+        headerName: "Fund",
+        field: "fund",
+        width: 200,
+        editable: true,
+        valueFormatter: (value) =>
+          !!value?.fund_name ? `${value?.fund_id} | ${value?.fund_name}` : "",
+        renderEditCell: (props) => (
+          <FundAutocompleteComponent {...props} data={data.moped_funds} />
+        ),
+      },
+      {
+        headerName: "Dept-unit",
+        field: "dept_unit",
+        width: 225,
+        editable: true,
+        valueFormatter: (value) =>
+          !!value?.unit_long_name
+            ? `${value?.dept} | ${value?.unit} |
+              ${value?.unit_long_name}`
+            : "",
+        renderEditCell: (props) => (
+          <FundingDeptUnitAutocomplete props={props} value={props.value} />
+        ),
+      },
+      {
+        headerName: "Amount",
+        field: "funding_amount",
+        width: 100,
+        editable: true,
+        valueFormatter: (value) => currencyFormatter.format(value),
+        renderEditCell: (props) => <DollarAmountIntegerField {...props} />,
+        type: "currency",
+      },
+      {
+        headerName: "",
+        field: "edit",
+        hideable: false,
+        filterable: false,
+        sortable: false,
+        editable: false,
+        type: "actions",
+        getActions: ({ id }) => {
+          const isInEditMode = rowModesModel[id]?.mode === GridRowModes.Edit;
+          if (isInEditMode) {
+            return [
+              <GridActionsCellItem
+                icon={<CheckIcon sx={{ fontSize: "24px" }} />}
+                label="Save"
+                sx={{
+                  color: "primary.main",
+                }}
+                onClick={handleSaveClick(id)}
+              />,
+              <GridActionsCellItem
+                icon={<CloseIcon sx={{ fontSize: "24px" }} />}
+                label="Cancel"
+                className="textPrimary"
+                onClick={handleCancelClick(id)}
+                color="inherit"
+              />,
+            ];
+          }
+          return [
+            <GridActionsCellItem
+              icon={<EditOutlinedIcon sx={{ fontSize: "24px" }} />}
+              label="Edit"
+              className="textPrimary"
+              onClick={handleEditClick(id)}
+              color="inherit"
+            />,
+            <GridActionsCellItem
+              icon={<DeleteOutlineIcon sx={{ fontSize: "24px" }} />}
+              label="Delete"
+              onClick={() => handleDeleteOpen(id)}
+              color="inherit"
+            />,
+          ];
+        },
+      },
+    ];
+  }, [
+    data,
+    rowModesModel,
+    handleDeleteOpen,
+    handleSaveClick,
+    handleCancelClick,
+    handleEditClick,
```

Comment:
i don't think the memoization is working here, because these handler functions are re-created on every render. wrapping them in useCallback() should do the trick 🤷

Reply:
I wrapped them in useCallback and added their dependencies. 

How do you check to see if they are being re-created on every render?

Reply:
good catch, John! yes, i see how these functions would have caused the memo to recalculate every render since their reference would update every render too.

as far as testing goes, i think you could log inside the memo and compare how often you see the logging or something similar. my short and lazy answer is because the docs say so. 🙃

Reply:
ditto 🙃  the way i tested this is by adding console logging within the hook and compared that to when i completely deleted the dependency array (square brackets and all)—when there is no dep array any hook will run on every render